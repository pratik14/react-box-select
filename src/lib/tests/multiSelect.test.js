import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import MultiSelect from '../multiSelect';

describe('<MultiSelect />', () => {
  let options;

  beforeEach(() => {
    let step;
    options = [];
    for (step = 1; step < 10; step += 1) {
      options.push({ id: `${step}`, label: `Element ${step}` });
    }
  });

  it('should render without crashing', () => {
    const tree = renderer.create(<MultiSelect options={options} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('#showSelectAllBtn', () => {
    const tree = renderer
      .create(<MultiSelect options={options} showSelectAllBtn={false} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Select/Unselect All event', () => {
    const wrapper = shallow(<MultiSelect options={options} showSelectAllBtn />);

    wrapper.find('#selectAll').simulate('click');

    let selectedOptions = wrapper.state().options.filter(o => o.selected);
    expect(selectedOptions.length).toEqual(options.length);

    wrapper.find('#unSelectAll').simulate('click');

    selectedOptions = wrapper.state().options.filter(o => o.selected);
    expect(selectedOptions.length).toEqual(0);
  });

  it('#handleClick', () => {
    const opt = options[0];
    const wrapper = shallow(<MultiSelect options={options} showSelectAllBtn />);

    let selectedOption = wrapper.state().options.find(o => opt.id === o.id);
    expect(selectedOption.selected).toEqual(undefined);

    wrapper.instance().handleClick(opt, 'selectable');

    selectedOption = wrapper.state().options.find(o => opt.id === o.id);
    expect(selectedOption.selected).toEqual(true);

    wrapper.instance().handleClick(opt, 'selected');
    selectedOption = wrapper.state().options.find(o => opt.id === o.id);
    expect(selectedOption.selected).toEqual(false);
  });
});
