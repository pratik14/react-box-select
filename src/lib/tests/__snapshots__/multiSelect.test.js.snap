// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiSelect /> #showSelectAllBtn 1`] = `
<div
  className="ms-container"
>
   
  <div />
  <div
    className="ms-selectable"
  >
    <ul
      className="ms-list"
    >
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 1
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 2
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 3
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 4
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 5
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 6
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 7
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 8
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 9
         
      </li>
    </ul>
  </div>
  <div
    className="ms-selected"
  >
    <ul
      className="ms-list"
    />
  </div>
   
</div>
`;

exports[`<MultiSelect /> should render without crashing 1`] = `
<div
  className="ms-container"
>
   
  <div />
  <div
    className="ms-selectable"
  >
    <ul
      className="ms-list"
    >
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 1
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 2
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 3
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 4
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 5
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 6
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 7
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 8
         
      </li>
      <li
        className="ms-elem-selectable"
        onClick={[Function]}
      >
        Element 9
         
      </li>
    </ul>
  </div>
  <div
    className="ms-selected"
  >
    <ul
      className="ms-list"
    />
  </div>
   
</div>
`;
